src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -289,7 +289,7 @@ internal bool TryWriteDataExtensionProperty(Utf8JsonWriter writer, T value, Json
 
             if (writer.CurrentDepth >= options.EffectiveMaxDepth)
             {
-                ThrowHelper.ThrowJsonException_SerializerCycleDetected(options.MaxDepth);
+                ThrowHelper.ThrowJsonException_SerializerCycleDetected(options.EffectiveMaxDepth);
             }
```

Comment:
It looks to me that this condition with never be true; the `JsonExtensionData` dictionary itself does not increase the `CurrentDepth` of the writer, the elements within does it and that is already covered in the `TryWrite` method.

My point here is that stepping into this branch is impossible and instead we should probably remove it.
Hence, having code coverage for this code path (https://github.com/dotnet/runtime/pull/34410/files#r401931623) does not apply after all.

@steveharter @layomia thoughts?

Reply:
This makes sense @Jozkee. Let's open a PR.